src/analysis/real/polynomial.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import data.set.intervals.infinite
+import topology.algebra.polynomial
+import analysis.calculus.mean_value
+
+/-!
+This file contains some lemmas about real polynomials and their derivatives.
+-/
+
+open polynomial real set
+
+
+lemma exists_forall_ge_of_polynomial_eval (α : ℝ) (f : polynomial ℝ):
+  ∃ M : ℝ, 0 < M ∧ ∀ (y : ℝ), abs (y - α) ≤ 1 → abs (eval y f) ≤ M :=
+begin
+  obtain ⟨x_max, ⟨h_x_max_range, hM⟩⟩ := is_compact.exists_forall_ge (@compact_Icc (α - 1) (α + 1))
+    ⟨α, le_of_lt $ sub_one_lt _, le_of_lt $ lt_add_one _⟩
+    (continuous_abs.comp f.continuous).continuous_on,
+  use [max (abs (f.eval x_max)) 1, lt_of_lt_of_le zero_lt_one (le_max_right _ _)],
+  intros y hy,
+  have hy' : y ∈ Icc (α - 1) (α + 1),
+  { apply mem_Icc.mpr,
+    have h1 := le_abs_self (y - α),
+    have h2 := neg_le_abs_self (y - α),
+    split; linarith },
+  exact le_trans (hM y hy') (le_max_left _ _),
+end
+
+lemma non_root_interval_of_polynomial (α : ℝ) (f : polynomial ℝ) (h_f_nonzero : f ≠ 0) :
```

Comment:
This should probably use something in #5298 (thas was made exactly because of this lemma I think). @adomani should be interested.

Reply:
Yes, #5298 is the first PR in a series that should provide a more topological approach, although this is not complete yet.  The later #5332 is still waiting for approval.  Yet, even with that PR there is still a little bit to be done.  From the PRs, the zero set of a (non-zero, depending on how you define stuff) polynomial is discrete.  Thus, you can separate each one of its singletons from the complement.  These are more or less one line away from the lemmas in the PRs.

Next, in a metric space, you can find pairwise disjoint open balls around each point of a finite subset (should be easy from the separation, going from open sets to balls).  The smallest radius of one such balls is the `B` asserted in `non_root_interval_of_polynomial`.

Not much is missing, but a little is.